src/VisualStudio/Core/Def/Implementation/ProjectSystem/RuleSets/VisualStudioRuleSetManager.RuleSetFile.cs

```diff
@@ -92,51 +103,62 @@ public ReportDiagnostic GetGeneralDiagnosticOption()
 
             private void EnsureSubscriptions()
             {
-                if (!_subscribed)
+                lock (_gate)
                 {
-                    foreach (var tracker in _trackers)
+                    if (!_subscribed)
                     {
-                        tracker.EnsureSubscription();
-                    }
+                        foreach (var tracker in _trackers)
```

Comment:
is this guranteed to be called after InitializeFileTracking?

Reply:
InitializeFileTracking is called before we ever hand out a rule set file, so yes.